### PR TITLE
Encrypted DataStore

### PR DESCRIPTION
**Pull request by EvgeniyFilippov**
## Summary
- Added an example of saving user data when registering in an encrypted storage
- Added Timber

### CODE REVIEW

File: app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    fun encrypt(bytes: ByteArray, outputStream: OutputStream): ByteArray {
+        val encryptedBytes = encryptCipher.doFinal(bytes)
+        outputStream.use {
+            it.write(encryptCipher.iv.size)
+            it.write(encryptCipher.iv)
+            it.write(encryptedBytes.size)
+            it.write(encryptedBytes)
+        }
+        return encryptedBytes
```
Comment (anastasiaShpakava) [inline app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt:58]: do we need handle exceptions?  (InvalidKeyException or BadPaddingException)

Comment (bstolyarov) [reply app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt:58]: @EvgeniyFilippov any updates here?

File: app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        return inputStream.use {
+            val ivSize = it.read()
+            val iv = ByteArray(ivSize)
+            it.read(iv)
+
+            val encryptedBytesSize = it.read()
+            val encryptedBytes = ByteArray(encryptedBytesSize)
+            it.read(encryptedBytes)
+
+            getDecryptCipherForIv(iv).doFinal(encryptedBytes)
```
[the remark this answers is not in the document]
Comment (bstolyarov) [reply app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt:71]: @EvgeniyFilippov any updates here?

[the remark this answers is not in the document]
Comment (EvgeniyFilippov) [reply app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt:71]: I think it's not necessary to be able to manually intercept and handle errors when writing/reading to try/catch blocks at the viewmodel level.

File: app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt (hunk 1 above)
Comment (EvgeniyFilippov) [reply app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt:58]: I think it's not necessary to be able to manually intercept and handle errors when writing/reading to try/catch blocks at the viewmodel level.

**Review by bstolyarov — APPROVED** [review]

**Review by anastasiaShpakava — APPROVED** [review]

**Pull request merged**